packages/lit-html/src/lit-html.ts

```diff
@@ -711,6 +715,48 @@ class Template {
   }
 }
 
+function resolveDirective(
+  part: NodePart | AttributePart,
+  value: unknown,
+  directiveParent: DirectiveParent = part,
+  attributeIndex?: number
+): unknown {
+  let currentDirective =
+    attributeIndex !== undefined
+      ? (directiveParent as AttributePart)._directives?.[attributeIndex]
+      : (directiveParent as NodePart | Directive)._directive;
+  const nextDirectiveConstructor = isPrimitive(value)
+    ? undefined
+    : (value as DirectiveResult)?._$litDirective$;
```

Comment:
`null` treated as a primitive in the `isPrimitive` check, so it's still not safe to dereference even when it passes. I added `isPrimitive` because it seemed to shave some time off the benchmarks, but I should double-check that.